docs/designs/xref.md

```diff
@@ -0,0 +1,128 @@
+    }
+outputs:
+  docs/a.json:
+  xrefmap.json: | 
+    {"references":[{"uid":"a","href":"docs/a.json","summary":"<pre><code>Hello `docfx`!\n</code></pre>\n"}]}
+```
+Since the xref map would be outputted for external reference, if markdown conatains a linking url, should it be resolved to an absolute url? It is resolve as related url for now.
+
+## Output xref map
+Docfx will output a JSON file named `xrefmap.json`. In V2, docfx used to output `xrefmap.yml`, it took much longer to be de-serialized compared to JSON format.
+```json
+{
+    "references": [
+        {
+            "uid": "a",
+            "href": "aspnet/a.md",
+            "name": "ASP.NET Documentation",
+            "fullName": "ASP.NET Full Documentation"
+        }
+    ]
+}
+```
+This file will be updated to some central place, and there would be a step to merge all the JSON files for the same share base path [Related issue](https://github.com/dotnet/docfx/issues/3487).
+
+## Create xref map
+There are two parts of xref map, internal and external.
+### Internal xref map
+User can define uids in the same repository and reference to them, and these defined uids will be outputted as `xrefmap.json`.
+
+### External xref map
+User can reference uids defined by other repositories as well, just define the urls of `xrefmap.json` in `docfx.yml`.
+```yml
+xref:
+  - http://url1/xrefmap.json
+  - http://url2/xrefmap.json
+```
+During `docfx restore`, all the JSON files will be restored and merged.
+
+## Resolve xref
+### Using `@` to reference a uid in markdown
+  Not supporting `displayProperty` for this case for now, we need to refine the validation pattern of uid firstly, could reference to [V2](https://dotnet.github.io/docfx/tutorial/links_and_cross_references.html#shorthand-form).
+```
+Link to @a
+```
+  The file above would be resolved using the restored xref map
+```json
+{"content":"<p>Link to <a href=\"aspnet/a.md\">ASP.NET Documentation</a></p>\n"}
+```
+  - User can also define which property to display for the referenced uid
+
+### Using `xref` to reference a uid in markdown
+```
+Link to <xref:a?displayProperty=fullName>
+```
+  And it will be resolved as below, notice that the display title is from `fullName` not `name`
+```json
+{"content":"<p>Link to <a href=\"aspnet/a.md\">ASP.NET Full Documentation</a></p>\n"}
+```
+### For SDP(JSON/YAML) files
+```json
+    {
+      "$schema": "https://raw.githubusercontent.com/dotnet/docfx/v3/schemas/TestData.json",
+      "xref": "a"
+    }
+```
+The expected resolved result would be a `XrefSpec` instance for the referenced uid. The remaining problem here is that we want to output an object with a string input at the same `xref` property [Related issue](https://github.com/dotnet/docfx/issues/3497).
+
+### Others
+There are some other UID reference formats supported in docfx V2, which are not supported in V3 yet:
+- [Markdown link style](https://dotnet.github.io/docfx/tutorial/links_and_cross_references.html#markdown-link)
```

Comment:
We may check BI data to see whether we need to do these items at this moment.

Reply:
As this is part of DFM spec, we should support this finally.

Reply:
why do we want to support DFM?

Reply:
I mean the DFM syntax instead of DFM engine. Markdig engine is expected to support all the DFM syntax.

Reply:
understood :)